File: Zds/Zds.Flow/Stopwatch/SystemStopwatch.vb
Namespace Stopwatch
    <DebuggerStepThrough>
    Public Class SystemStopwatch
        Implements IStopwatch
        Private _Stopwatch As New Diagnostics.Stopwatch
        Public ReadOnly Property IsRunning As Boolean Implements Interfaces.IStartStopable.IsRunning
            Get
                Return _Stopwatch.IsRunning
            End Get
        End Property
        Public Property Elapsed As TimeSpan Implements IStopwatch.Elapsed
            Get
                Return _Stopwatch.Elapsed
            End Get
            Set(value As TimeSpan)
                Throw New InvalidOperationException("You can assign this property to SystemStopwatch.")
            End Set
        End Property
        Public Sub Start() Implements Interfaces.IStartStopable.Start
            _Stopwatch.Start()
        End Sub
        Public Sub [Stop]() Implements Interfaces.IStartStopable.Stop
            _Stopwatch.Stop()
        End Sub
        Public Sub Reset() Implements Interfaces.IResetable.Reset
            _Stopwatch.Reset()
        End Sub
    End Class
End Namespace

File: Flow/Flow/Stopwatch/SystemStopwatch.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Flow.Stopwatch
{
    public class SystemStopwatch : IStopwatch
    {
        private readonly System.Diagnostics.Stopwatch stopwatch = new System.Diagnostics.Stopwatch();
        public bool IsRunning => stopwatch.IsRunning;
        public TimeSpan Elasped {
            get => stopwatch.Elapsed;
            set => throw new InvalidOperationException("You can assign this property to SystemStopwatch.");
        }
        public void Start() => stopwatch.Start();
        public void Stop() => stopwatch.Stop();
        public void Reset() => stopwatch.Reset();
    }
}
